### PR TITLE
feat: comprehensive testing audit — 51% to 76% coverage

### PR DESCRIPTION
**Pull request by ridermw**
## Summary

Comprehensive testing audit of NanoPilot. Coverage improved from **51.4%** to **75.6%** statement coverage with 471 tests (up from 255).

### New Test Files
- `src/mount-security.test.ts` — 46 tests for security-critical mount validation (3% → 97%)
- `src/index.test.ts` — 19 tests for main orchestrator (6% → 38%)
- `src/env.test.ts` — 14 tests for .env parsing (22% → 100%)
- `src/logger.test.ts` — 12 tests for logging (78% → 92%)
- `src/router.test.ts` — 21 tests for message routing (68% → 100%)
- `src/db-expanded.test.ts` — 27 tests for DB operations (72% → 85%)
- `src/ipc-update.test.ts` — 9 tests for update_task IPC
- `src/task-scheduler-expanded.test.ts` — 9 tests for scheduler loop
- `setup/platform-expanded.test.ts` — 32 tests for cross-platform detection (38% → 89%)
- `test/e2e/mocked.test.ts` — 7 E2E tests (full message lifecycle)
- `test/chaos/resilience.test.ts` — 15 chaos tests (resilience under hostile conditions)

### Expanded Test Files
- `src/container-runner.test.ts` — expanded from 5 to 15 tests (68% → 82%)

### Infrastructure
- `vitest.e2e.config.ts` — separate E2E test config with 120s timeout
- `test/helpers/mocks.ts` — shared mock factories (DRY)
- `test/helpers/README.md` — DB testing strategy docs

### Source Changes (test support only)
- `src/mount-security.ts` — added `_resetMountSecurityForTests()` export
- `src/index.ts` — added 10 `_` prefix exports for internal functions (matches existing pattern)
- `.gitignore` — added coverage/ exclusion

## Coverage Summary

| File | Before | After |
|------|--------|-------|
| env.ts | 22% | **100%** |
| mount-security.ts | 3% | **97%** |
| router.ts | 68% | **100%** |
| logger.ts | 78% | **92%** |
| platform.ts | 38% | **89%** |
| db.ts | 72% | **85%** |
| container-runner.ts | 68% | **82%** |
| task-scheduler.ts | 45% | **78%** |
| **Overall** | **51.4%** | **75.6%** |

## Remaining Gaps (future work)
- `index.ts` (38%) — needs more orchestrator integration tests
- `ipc.ts` (63%) — `startIpcWatcher` filesystem polling is hard to unit test
- `group-queue.ts` (84%) — async path edge cases

## Test Commands
```bash
npm test                                    # Unit tests (464 tests)
npx vitest run --config vitest.e2e.config.ts  # E2E tests (7 tests)
npx vitest run --coverage                   # With coverage report
```

### CODE REVIEW

File: src/logger.test.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      .mockImplementation(() => true);
+  });
+
+  afterEach(() => {
+    stdoutWrite.mockRestore();
+    stderrWrite.mockRestore();
+    if (originalLogLevel !== undefined) {
+      process.env.LOG_LEVEL = originalLogLevel;
+    }
+  });
```
[the remark this answers is not in the document]
Comment (ridermw) [reply src/logger.test.ts]: Fixed in b36871a — `beforeEach` now saves `originalLogLevel` and `afterEach` deletes `LOG_LEVEL` when it was originally unset.

File: src/task-scheduler-expanded.test.ts [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+} from './task-scheduler.js';
+import { ScheduledTask } from './types.js';
+
+function createDeps(
+  overrides: Partial<SchedulerDependencies> = {},
+): SchedulerDependencies {
+  return {
+    registeredGroups: () => ({
+      'chat-1': { name: 'G1', folder: 'g1', trigger: '@Bot', isMain: false, added_at: new Date().toISOString() },
+    }),
```
[the remark this answers is not in the document]
Comment (ridermw) [reply src/task-scheduler-expanded.test.ts]: Already fixed in 344d7c4 — `added_at` added to RegisteredGroup fixture.

**Comment by ridermw** [comment]
## Test Suite Consolidation (commit d14f018)

Critic-validated audit removed **41 redundant tests** with zero coverage loss:

| Action | Tests Removed | Tests Added/Moved | Net |
|--------|:---:|:---:|:---:|
| Delete `router.test.ts` → migrate 6 to `formatting.test.ts` | -21 | +6 | -15 |
| Trim `platform.test.ts` to 5 smoke tests + add isRoot to expanded | -13 | +7 | -6 |
| Move 3 `computeNextRun` to expanded, keep integration test | -3 | +3 | 0 |
| Trim E2E from 7 to 1 comprehensive test | -6 | 0 | -6 |
| Trim chaos from 15 to 6 unique boundary tests | -9 | 0 | -9 |
| Remove 3 duplicate task CRUD from `db-expanded` | -3 | 0 | -3 |
| **Redundant** → _removed_ | | | |
| **Unique** → _relocated or kept_ | | | |

**Before:** 471 tests, 75.6% coverage
**After:** 430 tests, 75.59% coverage ✅

Key decisions from critic review:
- Kept `platform.test.ts` as a slim smoke file (real-environment tests can't live in mocked file)
- Kept scheduler integration test (real DB, not just mock assertions)
- Enhanced E2E test to cover multi-channel routing + content escaping in one test
- All TypeScript clean (`tsc --noEmit` passes)

File: src/task-scheduler-expanded.test.ts [hunk 3]
```diff
@@ -0,0 +1,389 @@
+/**
+ * Expanded tests for task-scheduler.ts — covers runTask() and startSchedulerLoop().
+ * Existing task-scheduler.test.ts covers computeNextRun().
```
[the remark this answers is not in the document]
Comment (ridermw) [reply src/task-scheduler-expanded.test.ts]: Fixed in ee469bf.
